lumigator/python/mzai/backend/backend/services/workflows.py

```diff
@@ -0,0 +1,166 @@
+                )
+
+    def _run_eval(
+        self,
+        inference_job_id: UUID,
+        request: WorkflowCreate,
+        background_tasks: BackgroundTasks,
+        experiment_id: UUID = None,
+    ):
+        # use the inference job id to recover the dataset record
+        dataset_record = self._dataset_service._get_dataset_record_by_job_id(inference_job_id)
+
+        # prepare the inputs for the evaluation job and pass the id of the new dataset
+        job_eval_dict = {
+            "name": f"{request.name}-evaluation",
+            "model": request.model,
+            "dataset": dataset_record.id,
+            "max_samples": request.max_samples,
+            "skip_inference": True,
+        }
+
+        # submit the job
+        self._job_service.create_job(
+            JobEvalLiteCreate.model_validate(job_eval_dict),
+            background_tasks,
+            experiment_id=experiment_id,
+        )
+
+    def create_workflow(
+        self, request: WorkflowCreate, background_tasks: BackgroundTasks
+    ) -> WorkflowResponse:
+        """Creates a new workflow and submits inference and evaluation jobs.
+
+        Args:
+            request (WorkflowCreate): The request object containing the workflow configuration.
+            background_tasks (BackgroundTasks): The background tasks manager for scheduling tasks.
+
+        Returns:
+            WorkflowResponse: The response object containing the details of the created workflow.
+        """
+        loguru.logger.info(
+            f"Creating workflow '{request.name}' for experiment ID '{request.experiment_id}'."
+        )
+
+        # input is WorkflowCreate, we need to split the configs and generate one
+        # JobInferenceCreate and one JobEvalCreate
+        job_inference_dict = {
+            "name": f"{request.name}-inference",
+            "model": request.model,
+            "dataset": request.dataset,
+            "max_samples": request.max_samples,
+            "model_url": request.model_url,
+            "output_field": request.inference_output_field,
+            "system_prompt": request.system_prompt,
+            "store_to_dataset": True,
+        }
+
+        # submit inference job first
+        job_response = self._job_service.create_job(
+            JobInferenceCreate.model_validate(job_inference_dict),
+            background_tasks,
+            experiment_id=request.experiment_id,
+        )
+
+        # run evaluation job afterwards
+        # (NOTE: tasks in starlette are executed sequentially: https://www.starlette.io/background/)
+        background_tasks.add_task(
+            self.on_job_complete,
+            job_response.id,
+            self._run_eval,
+            job_response.id,
+            request,
+            background_tasks,
+            request.experiment_id,
+        )
+
+        # TODO create a new workflow object which will be stored in
+        # by the tracking service (aka mlflow)
+        created_at = datetime.datetime.now()
```

Comment:
nit, but may want to specify `datetime.datetime.utcnow()` for consistency across machines, for debugging, etc

Reply:
Yeah this was really just a placeholder for this PR, it's going away in the next one https://github.com/mozilla-ai/lumigator/pull/750